purchases.go

```diff
@@ -18,5 +18,5 @@ type Purchase struct {
 	CustomerNotes         string            `xml:"customer_notes,omitempty"`
 	TermsAndConditions    string            `xml:"terms_and_conditions,omitempty"`
 	VATReverseChargeNotes string            `xml:"vat_reverse_charge_notes,omitempty"`
-	ShippingAddressID     int               `xml:"shipping_address_id,omitempty"`
+	ShippingAddressID     string            `xml:"shipping_address_id,omitempty"`
```

Comment:
@nelz9999 Recurly documents this as a `long` not a `string`. This should probably be an `int64` type (although I'm not sure what specific error you were getting). Also please replicate with a unit test using Recurly's sample values.

Reply:
Blargh... 😭 

This is where Recurly is a bit crazy-making... Their docs say `string` on other endpoints: https://dev.recurly.com/docs/edit-shipping-address-on-an-existing-account

So, I can see an argument for `int64`, but for consistency and interoperability w/ the `ShippingAddressService` which models the `ID` as a `string`, I'd suggest that we continue with the pattern of `string`.

I'll go with that, but totes open to pushback.

Reply:
I'm not a fan of how Recurly documents their API as it's (among other things) a bit confusing.

If you look here in the sidebar (the XML response): you can see the shipping id as `type="integer"`
https://dev.recurly.com/docs/list-accounts-shipping-address
https://dev.recurly.com/docs/create-shipping-address-on-an-account

If you look here it's documented as a string, but that's because it's a URL param in this context:
https://dev.recurly.com/docs/edit-shipping-address-on-an-existing-account

The purchase API is where it's documented as the specific type of integer (a `long`) which matches up with their XML examples:
https://dev.recurly.com/docs/create-purchase

You see them use the same style on their gift cards API.

Create gift card shows a response XML of `type="id"`
https://dev.recurly.com/docs/create-gift-card

Lookup gift card documents the id (url param) as a string, but in the response you see it's an integer:
https://dev.recurly.com/docs/lookup-a-gift-card

Since we are using XML to marshal/unmarshal, the URL params aren't relevant as they'll all be sent as strings. I'd prefer we use int64 so we retain the correct types and make it easier for people who want to store the correct types in a local database. This PR should update the `ShippingAddress` service as well.

I haven't used the API directly -- just going off of the docs -- but it sounds like you're doing some integration testing with their API. Do those responses line up up with an `int64` type like what their docs show? If so let's have the test return an `int` that would exceed the bounds of an `int` type so we aren't just testing with an int like `1234` that could be used as either an `int` or `int64` without error. I pulled this one from their docs: `2438622711411416831`

Appreciate all the contributions on this. Let me know if anything in your live testing with their API doesn't match up with an `int64` type. I'm happy to reach out to Recurly support directly as well if we end up needing further clarification!